src/RadixDlt.NetworkGateway.PostgresIntegration/Interceptors/MetricsInterceptor.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Storage.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using RadixDlt.NetworkGateway.GatewayApi.Configuration;
+using RadixDlt.NetworkGateway.GatewayApi.Services;
+using RadixDlt.NetworkGateway.PostgresIntegration.Metrics;
+using System.Data.Common;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace RadixDlt.NetworkGateway.PostgresIntegration.Interceptors;
+
+internal class MetricsInterceptor : DbCommandInterceptor
+{
+    private const string UnknownQueryName = "UNKNOWN";
+    private readonly IOptionsMonitor<SlowQueriesLoggingOptions> _slowQueriesLoggingOptions;
+    private readonly ILogger<MetricsInterceptor> _logger;
+    private readonly ISqlQueryObserver _sqlQueryObserver;
+
+    public MetricsInterceptor(ILoggerFactory loggerFactory, IOptionsMonitor<SlowQueriesLoggingOptions> slowQueriesLoggingOptions, ISqlQueryObserver sqlQueryObserver)
+    {
+        _slowQueriesLoggingOptions = slowQueriesLoggingOptions;
+        _sqlQueryObserver = sqlQueryObserver;
+        _logger = loggerFactory.CreateLogger<MetricsInterceptor>();
```

Comment:
Why don't we just inject `ILogger<MetricsInterceptor>`?

Reply:
good catch, forgot to change that.